easybuild/easyconfigs/s/SpFFT/SpFFT-0.9.8-CrayGNU-19.10-cuda-10.1.eb

```diff
@@ -0,0 +1,36 @@
+easyblock = 'CMakeMake'
+
+name = 'SpFFT'
+version = '0.9.8'
+cudaversion = 10.1
+versionsuffix = '-cuda-%s' % cudaversion
+
+homepage = 'https://github.com/eth-cscs/SpFFT'
+description = """Sparse 3D FFT library with MPI, OpenMP, CUDA and ROCm support"""
+
+toolchain = {'name': 'CrayGNU', 'version': '19.10'}
+toolchainopts = {'usempi': True, 'openmp': True, 'pic': True, 'verbose': False, 'opt': True}
+
+source_urls = ['https://github.com/eth-cscs/SpFFT/archive']
+sources = ['v%(version)s.tar.gz']
+
+builddependencies = [('CMake', '3.14.5', '', True)]
+
+dependencies = [
+  ('cudatoolkit' , EXTERNAL_MODULE),
```

Comment:
We updated the EB toolchains and, since Cray PE 19.10, you don't need to pin the version for the modules listed in the cray `cdt/19.10` module.

However,  `cudatoolkit` is not in the list and therefore we recommend to pin the cudatoolkit version here (for reproducibility, to ensure that this recipe will work when we change the cuda defaults on the system).

Reply:
Ok.

Reply:
So, then you need to have different EB recipes for each version of CUDA?

Reply:
It was always the case to have different recipes for different versions of cuda: this is how reproducibility is guaranteed, we cannot rely on system defaults as they might change over time and break the existing recipes. 

The only thing we changed recently is that we include the cdt module in the toolchains, so that you don't need to explicitly pin (most of) the Cray dependency versions in the recipes and you will automatically get the default version from the specific cdt. The defaults for a specific cdt are defined by Cray and they don't change over time (this should not be confused with the system defaults: loading cdt will override system defaults).

The question 'what is special about cudatoolkit' should be asked to Cray, since they don't define a default for cuda in the cdt module (see for example `/opt/cray/pe/modulefiles/cdt/19.10).

We missed this detail when we deployed the 19.10 software stack (that's why you can see versionless dependencies on cudatoolkit in other recipes), but we'll fix that soon.